### PR TITLE
 #2883: Non Federal Agency Update - [RH] 

### LINKED ISSUE #2883: Updating anything on organization name/address page results in Agency type changing
### Current Behavior

When a user of a non-federal domain edits any organization name/address info in the domain management pages, the `Federal agency` field changes from "Non-Federal Agency" to "None". (The field exists for all users but is not displayed to non-federal domains, though the field does show up in [current-full/federal](https://github.com/cisagov/dotgov-data/blob/main/current-full.csv#L5787).)

### Expected Behavior

The `Federal agency` field should **never** be anything but None for domains, and non-federal domains should show as "Non-Federal Agency".

### Steps to Reproduce

1. Login to any system and click on a **non-federal** domain you manage (in dns needed, unknown or Ready state).
2. Go to the Organization name and address page
3. Edit anything in the name and address
4. Click save
5. Go to the audit logs and notice that in addition to the changes you made, the federal agency of Non federal has changed to None

### Environment

any

### Additional Context

[slack thread with more context](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1727373330241409)

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #2883 

## Changes

- Making sure we set the Federal Agency field to Non-Federal Agency and not set it to None every time we make a change in Organization in the Domain Overview

## Context for reviewers

- When we changed any fields in the Organization for the first time in Domain Overview, it'd also set our Federal Agency to None

## Setup
1. Go to my sandbox and create a new Domain Request and then go through the process of approving it
2. Once approved, go to your Domain and update the address or any other field in the Organization fields and hit save
3. Go to Log Entries in /admin and the Federal Agency should NOT be changed (there should be no change in federal_agency field just the change you made to whichever field you decided to update)  -- see screenshot

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots
Screenshot 2024-10-07 at 2 38 24 PM

### CODE REVIEW

File: src/registrar/admin.py [hunk 1]
```diff
@@ -2464,7 +2464,10 @@ def generic_org_type(self, obj):
     generic_org_type.admin_order_field = "domain_info__generic_org_type"  # type: ignore
 
     def federal_agency(self, obj):
-        return obj.domain_info.federal_agency if obj.domain_info else None
+        if obj.domain_info:
```
Comment (therealslimhsiehdy) [inline src/registrar/admin.py:2467]: FYI I did not change this, lint did!

Comment (therealslimhsiehdy) [inline src/registrar/admin.py:2467]: FYI this statement was changed by of the linter, not me!

File: src/registrar/forms/domain.py [hunk 2]
```diff
@@ -529,17 +529,25 @@ def __init__(self, *args, **kwargs):
...
         """Override the save() method of the BaseModelForm."""
+
         if self.has_changed():
 
             # This action should be blocked by the UI, as the text fields are readonly.
             # If they get past this point, we forbid it this way.
             # This could be malicious, so lets reserve information for the backend only.
-            if self.is_federal and not self._field_unchanged("federal_agency"):
-                raise ValueError("federal_agency cannot be modified when the generic_org_type is federal")
+
```
Comment (therealslimhsiehdy) [inline src/registrar/forms/domain.py:538]: FYI this statement was also changed by of the linter, not me!

**Review by asaki222 — APPROVED** [review]
LGTM!

**Pull request merged**